apps/speed.c

```diff
@@ -3161,6 +3554,369 @@ int speed_main(int argc, char **argv)
+                           sig_name);
+                goto sig_err_break;
+            }
+            if (EVP_PKEY_keygen(sig_gen_ctx, &pkey) <= 0) {
+                BIO_printf(bio_err,
+                           "Error while generating signature EVP_PKEY for %s.\n",
+                           sig_name);
+                goto sig_err_break;
+            }
+            /* Now prepare signature data structs */
+            sig_sign_ctx = EVP_PKEY_CTX_new_from_pkey(app_get0_libctx(),
+                                                      pkey,
+                                                      app_get0_propq());
+            if ((sig_sign_ctx == NULL)
+                || (EVP_PKEY_sign_init(sig_sign_ctx) <= 0)
+                || ((strncmp(sig_name, "rsa", 3) == 0
+                  && (EVP_PKEY_CTX_set_rsa_padding(sig_sign_ctx,
+                                               RSA_PKCS1_PADDING)  <= 0)))
+                || (EVP_PKEY_sign(sig_sign_ctx, NULL, &max_sig_len,
+                                      md, md_len) <= 0)) {
+                BIO_printf(bio_err,
+                           "Error while initializing signing data structs for %s.\n",
+                           sig_name);
+                goto sig_err_break;
+            }
+            sig = OPENSSL_malloc(sig_len = max_sig_len);
+            if (sig == NULL) {
+                BIO_printf(bio_err, "MemAlloc error in sign for %s.\n", sig_name);
+                goto sig_err_break;
+            }
+            if (EVP_PKEY_sign(sig_sign_ctx, sig, &sig_len,
+                                     md, md_len) <= 0) {
+                BIO_printf(bio_err, "Signing error for %s.\n", sig_name);
+                goto sig_err_break;
+            }
+            /* Now prepare verify data structs */
+            memset(md, 0, SHA256_DIGEST_LENGTH);
+            sig_verify_ctx = EVP_PKEY_CTX_new_from_pkey(app_get0_libctx(),
+                                                        pkey,
+                                                        app_get0_propq());
+            if ((sig_verify_ctx == NULL)
+                || (EVP_PKEY_verify_init(sig_verify_ctx) <= 0)
+                || ((strncmp(sig_name, "rsa", 3) == 0
+                  && (EVP_PKEY_CTX_set_rsa_padding(sig_verify_ctx,
+                                               RSA_PKCS1_PADDING) <= 0)))) {
+                BIO_printf(bio_err,
+                           "Error while initializing verify data structs for %s.\n",
+                           sig_name);
+                goto sig_err_break;
+            }
+            if (EVP_PKEY_verify(sig_verify_ctx, sig, sig_len,
+                                      md, md_len) <= 0) {
+                BIO_printf(bio_err, "Verify error for %s.\n", sig_name);
+                goto sig_err_break;
+            }
+            if (EVP_PKEY_verify(sig_verify_ctx, sig, sig_len,
+                                      md, md_len) <= 0) {
```

Comment:
nit: wrong indent

Reply:
corrected in https://github.com/openssl/openssl/pull/19968/commits/7b25a24f195e11f45dd2d8c1f94e80fde36d4360